### PR TITLE
Fix Template and ISO upload events

### PR DESCRIPTION
**Pull request by nvazquez**
### Description

This PR adds the missing events when uploading templates and ISOs.
Fixes: #8052 

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI
- [ ] test (unit or integration test code)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### Screenshots (if appropriate):
Screenshot 2024-08-19 at 11 48 36

### How Has This Been Tested?
Tested uploading templates, ISOs and volumes for comparisson

### CODE REVIEW

**Comment by nvazquez** [comment]
@blueorangutan package

File: server/src/main/java/com/cloud/template/HypervisorTemplateAdapter.java [hunk 1]
```diff
@@ -359,6 +361,16 @@ public List<TemplateOrVolumePostUploadCommand> doInTransaction(TransactionStatus
...
+                // Set Event Details for Template/ISO Upload
+                String eventType = template.getFormat().equals(ImageFormat.ISO) ? "Iso" : "Template";
+                String eventResourceId = template.getUuid();
+                CallContext.current().setEventDetails(String.format("%s Id: %s", eventType, eventResourceId));
+                CallContext.current().putContextParameter(eventType.equals("Iso") ? eventType : VirtualMachineTemplate.class, eventResourceId);
+                if (template.getFormat().equals(ImageFormat.ISO)) {
+                    CallContext.current().setEventResourceType(ApiCommandResourceType.Iso);
+                    CallContext.current().setEventResourceId(template.getId());
+                }
+
```
Comment (Pearl1594) [inline server/src/main/java/com/cloud/template/HypervisorTemplateAdapter.java:373]: Is this spe

File: server/src/main/java/com/cloud/template/HypervisorTemplateAdapter.java [hunk 2]
```diff
@@ -359,6 +361,16 @@ public List<TemplateOrVolumePostUploadCommand> doInTransaction(TransactionStatus
                     throw new CloudRuntimeException("Unable to persist the template " + profile.getTemplate());
                 }
 
+                // Set Event Details for Template/ISO Upload
+                String eventType = template.getFormat().equals(ImageFormat.ISO) ? "Iso" : "Template";
+                String eventResourceId = template.getUuid();
+                CallContext.current().setEventDetails(String.format("%s Id: %s", eventType, eventResourceId));
+                CallContext.current().putContextParameter(eventType.equals("Iso") ? eventType : VirtualMachineTemplate.class, eventResourceId);
+                if (template.getFormat().equals(ImageFormat.ISO)) {
```
Comment (Pearl1594) [inline server/src/main/java/com/cloud/template/HypervisorTemplateAdapter.java:369]: Is this required to be specific to only ISOs?

**Review by Pearl1594 — APPROVED** [review]

File: server/src/main/java/com/cloud/template/HypervisorTemplateAdapter.java (hunk 2 above)
Comment (nvazquez) [reply server/src/main/java/com/cloud/template/HypervisorTemplateAdapter.java:369]: Yes, otherwise the event resource was not being obtained

**Comment by nvazquez** [comment]
@blueorangutan test

**Review by kiranchavala — APPROVED** [review]
LGTM

The events are present when a template and a iso is uploaded 

iso-upload

template-upload

**Pull request merged**